### PR TITLE
Add support for thumbnails in AASXSerializer / AASXDeserializer

### PR DESCRIPTION
**Pull request by fvolz**
The AASXSerializer/Deserializer ignored thumbnails when loading or saving files in the AASX.
The thumbnails are now properly processed as files.

### CODE REVIEW

File: dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXDeserializer.java [hunk 1]
```diff
@@ -148,6 +149,9 @@ private List<String> parseReferencedFilePathsFromAASX() throws IOException, Inva
         read();
 
         List<String> paths = new ArrayList<>();
+        for (AssetAdministrationShell aas : environment.getAssetAdministrationShells()) {
+            paths.add(aas.getAssetInformation().getDefaultThumbnail().getPath());
```
Comment (mjacoby) [inline dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXDeserializer.java:153]: Should probably add null checks for `aas.getAssetInformation()`, `getAssetInformation().getDefaultThumbnail()`, and `aas.getAssetInformation().getDefaultThumbnail().getPath()`.

File: dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXSerializer.java [hunk 2]
```diff
@@ -110,23 +111,33 @@ public void write(Environment environment, Collection<InMemoryFile> files, Outpu
...
      * @param rootPackage the OPCPackage
      * @param xmlPart the Part the files should be related to
      */
-    private void storeFilesInAASX(List<Submodel> submodelList, Collection<InMemoryFile> files, OPCPackage rootPackage,
+    private void storeFilesInAASX(Environment environment, Collection<InMemoryFile> files, OPCPackage rootPackage,
                                   PackagePart xmlPart) {
-
-        for (Submodel sm : submodelList) {
+        for (AssetAdministrationShell aas : environment.getAssetAdministrationShells()) {
+            String filePath = AASXUtils.getPathFromURL(aas.getAssetInformation().getDefaultThumbnail().getPath());
```
Comment (mjacoby) [inline dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXSerializer.java:130]: Should probably add null checks for `aas.getAssetInformation()`, `getAssetInformation().getDefaultThumbnail()`, and `aas.getAssetInformation().getDefaultThumbnail().getPath()`.

File: dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXSerializer.java [hunk 3]
```diff
@@ -110,23 +112,37 @@ public void write(Environment environment, Collection<InMemoryFile> files, Outpu
...
                                   PackagePart xmlPart) {
-
-        for (Submodel sm : submodelList) {
+        for (AssetAdministrationShell aas : environment.getAssetAdministrationShells()) {
+            AssetInformation information = aas.getAssetInformation();
+            if(information == null || information.getDefaultThumbnail() == null || information.getDefaultThumbnail().getPath() == null) {
+                continue;
+            }
+            String filePath = AASXUtils.getPathFromURL(aas.getAssetInformation().getDefaultThumbnail().getPath());
+            try {
```
Comment (FrankSchnicke) [inline dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXSerializer.java:136]: This part is copy & paste of the logic in the Submodel processing. Please extract it in a common function

File: dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXSerializer.java [hunk 4]
```diff
@@ -110,23 +112,37 @@ public void write(Environment environment, Collection<InMemoryFile> files, Outpu
...
      * @param files the content of the files
      * @param rootPackage the OPCPackage
      * @param xmlPart the Part the files should be related to
      */
-    private void storeFilesInAASX(List<Submodel> submodelList, Collection<InMemoryFile> files, OPCPackage rootPackage,
+    private void storeFilesInAASX(Environment environment, Collection<InMemoryFile> files, OPCPackage rootPackage,
                                   PackagePart xmlPart) {
-
-        for (Submodel sm : submodelList) {
+        for (AssetAdministrationShell aas : environment.getAssetAdministrationShells()) {
```
Comment (FrankSchnicke) [inline dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXSerializer.java:130]: I think this whole logic could be drastically simplified by using a stream, using the predicate given in l. 132 as filter and then applying the rest of the logic via forEach

File: dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXDeserializer.java [hunk 5]
```diff
@@ -148,6 +150,13 @@ private List<String> parseReferencedFilePathsFromAASX() throws IOException, Inva
         read();
 
         List<String> paths = new ArrayList<>();
+        for (AssetAdministrationShell aas : environment.getAssetAdministrationShells()) {
+            AssetInformation information = aas.getAssetInformation();
+            if(information == null || information.getDefaultThumbnail() == null || information.getDefaultThumbnail().getPath() == null) {
```
Comment (FrankSchnicke) [inline dataformat-aasx/src/main/java/org/eclipse/digitaltwin/aas4j/v3/dataformat/aasx/AASXDeserializer.java:155]: See the other comment above. I think filtering via a stream and then using forEach is a better pattern in contrast to having a _continue_ within a loop

**Review by FrankSchnicke — CHANGES_REQUESTED** [review]
Thank you very much for your contribution. Please see my separate comments for the review.

**Review by FrankSchnicke — APPROVED** [review]

**Pull request merged**